### PR TITLE
Arc script updates and bugfixes

### PR DESCRIPTION
**Pull request by segeljakt**
This PR fixes some issues which in result makes more tests pass. This can possibly be useful for MLIR.

### CODE REVIEW

File: arc-script/arc-script-core/src/compiler/ast/display.rs [hunk 1]
```diff
@@ -396,7 +396,6 @@ impl<'i> Display for Pretty<'i, ast::BinOp, Context<'_>> {
             ast::BinOpKind::Leq  => write!(f, " <= "),
             ast::BinOpKind::Or   => write!(f, " or "),
             ast::BinOpKind::And  => write!(f, " and "),
-            ast::BinOpKind::Xor  => write!(f, " xor "),
```
Comment (frej) [inline arc-script/arc-script-core/src/compiler/ast/display.rs:399]: Actually we could support xor for booleans quite simply if you want it. The code generator would just transform  `(A xor B)` to `((A == !B) || (!A == B))`

**Review by frej — APPROVED** [review]
LGTM, the only thing is if you really want do drop xor for bools.

File: arc-script/arc-script-core/src/compiler/ast/display.rs (hunk 1 above)
Comment (frej) [reply arc-script/arc-script-core/src/compiler/ast/display.rs:399]: Or, as booleans are `i1` in MLIR, you could just use the bxor, the result is the same.

Comment (segeljakt) [reply arc-script/arc-script-core/src/compiler/ast/display.rs:399]: Interesting, then I might add it back.

Comment (segeljakt) [reply arc-script/arc-script-core/src/compiler/ast/display.rs:399]: Is it `arc.xor` or `std.bxor`?

Comment (frej) [reply arc-script/arc-script-core/src/compiler/ast/display.rs]: `std.xor`: https://mlir.llvm.org/docs/Dialects/Standard/#stdxor-xorop

File: arc-script/arc-script-test/insta/src/expect-pass/binops.arc [hunk 2]
```diff
@@ -265,6 +265,8 @@ fun main() {
 #CHECK: {{%[^ ]+}} = and {{%[^ ]+}}, {{%[^ ]+}} : i1
     let lor: bool = c_bool or c_bool in
 #CHECK: {{%[^ ]+}} = or {{%[^ ]+}}, {{%[^ ]+}} : i1
+    let lxor: bool = c_bool xor c_bool in
+#CHECK: {{%[^ ]+}} = arc.xor {{%[^ ]+}}, {{%[^ ]+}} : i1
```
Comment (frej) [inline arc-script/arc-script-test/insta/src/expect-pass/binops.arc:269]: It should be std.xor, arc.xor is not defined for `i1`, only `uiX` and `siX`.

**Review by frej — CHANGES_REQUESTED** [review]

**Review by frej — APPROVED** [review]

**Pull request merged**